### PR TITLE
Migrate adaptive_experimental to NNBD.

### PR DESCRIPTION
**Pull request by pennzht**
This PR migrates the package `adaptive_experimental` to non-null by default.

### CODE REVIEW

File: adaptive_experimental/lib/navigation_rail.dart [hunk 1]
```diff
@@ -105,9 +105,9 @@ class _NavigationRailState extends State<NavigationRail> with TickerProviderStat
   Widget build(BuildContext context) {
     final leading = widget.leading;
     return DefaultTextStyle(
-      style: TextStyle(color: Theme.of(context).colorScheme.primary),
+      style: TextStyle(color: Theme.of(context)!.colorScheme.primary),
```
Comment (guidezpl) [inline adaptive_experimental/lib/navigation_rail.dart:108]: nit: could you factor out `colorScheme`?

File: adaptive_experimental/lib/navigation_rail.dart [hunk 2]
```diff
@@ -230,7 +230,7 @@ class _RailItem extends StatelessWidget {
     }
 
     final colors = Theme
```
Comment (guidezpl) [inline adaptive_experimental/lib/navigation_rail.dart:232]: nit: could also factor out `colorScheme` here

**Review by clocksmith — APPROVED** [review]

**Review by JoseAlba — APPROVED** [review]
LGTM

File: adaptive_experimental/lib/navigation_rail.dart [hunk 3]
```diff
@@ -164,12 +164,12 @@ class _RailItem extends StatelessWidget {
...
   }
 
-  Animation _positionAnimation;
+  late Animation _positionAnimation;
 
   final Animation<double> animation;
   final NavigationRailKind labelKind;
   final bool selected;
-  final Icon icon;
+  final Widget icon;
```
Comment (guidezpl) [inline adaptive_experimental/lib/navigation_rail.dart:172]: this doesn't require a change anywhere else?

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Review by rami-a — APPROVED** [review]

File: adaptive_experimental/lib/navigation_rail.dart (hunk 3 above)
Comment (pennzht) [reply adaptive_experimental/lib/navigation_rail.dart:172]: `icon` is only used as the child of `SizedBox` and a child of `Column`, so any `Widget` can satisfy. Moreover, the following code requires `icon` to be a `Widget`:

```dart
                icon: widget.currentIndex == i
                    ? widget.items[i].activeIcon
                    : widget.items[i].icon,
```

(Lines 128–130 on `develop` in file `navigation_rail.dart`)

File: adaptive_experimental/lib/navigation_rail.dart (hunk 1 above)
Comment (pennzht) [reply adaptive_experimental/lib/navigation_rail.dart:108]: Good idea!

Comment (pennzht) [reply adaptive_experimental/lib/navigation_rail.dart]: Done.

File: adaptive_experimental/lib/navigation_rail.dart (hunk 2 above)
Comment (pennzht) [reply adaptive_experimental/lib/navigation_rail.dart]: Done.

**Pull request merged**